infrastructure/src/main/java/org/corfudb/infrastructure/log/StreamLogFiles.java

```diff
@@ -128,7 +130,26 @@ private long getStartingSegment() {
      * Create stream log directory if not exists
      */
     private void initStreamLogDirectory() {
+        // If FileSystem is mounted as read-only, Corfu server cannot function.
+        try {
+            FileStore fs = Files.getFileStore(Paths.get(logDir.getParent().toString()));
```

Comment:
Even if the directory does not exist, it should be able to retrieve the filesystem that it is mounted on which can then be used to test for permissions. Please correct me if I am mistaken.

Reply:
It will actually fail with an `java.nio.file.NoSuchFileException` , but throw `UnrecoverableCorfuError("Unable to retrieve Corfu Filesystem permissions")` 

Also, on failure, please print the `logDir` its more useful.

Reply:
done. fixed.

Reply:
Actually, it doesn't matter, because both would fail the initialization.